### PR TITLE
add delete org before run virt-who config upgrade cases

### PR DESCRIPTION
**Pull request by yanpliu**
Because ORG_DATA = {'name': 'virtwho_upgrade_org_name'} exist will effect the pre-upgrade case, so clean it before run the case, not use the dynamic name ,because post-upgrade will also this name
case : PASS
```
(robottelo_vv) [virtwho@dell-per740-68-vm-04 robottelo]$ pytest ./tests/upgrades/test_virtwho.py  -m pre_upgrade --disable-pytest-warnings
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.9, pytest-7.3.2, pluggy-1.0.0
Mandatory Requirements are up to date.
Optional Requirements are up to date.
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/virtwho/repository/robottelo
configfile: pyproject.toml
plugins: cov-4.1.0, ibutsu-2.2.4, xdist-3.3.1, mock-3.11.1, services-2.2.1, reportportal-5.1.9
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_virtwho.py .                                                                                                                                                                            [100%]

============================================================================ 1 passed, 1 deselected, 25 warnings in 131.65s (0:02:11) =============================================================================
(robottelo_vv) [virtwho@dell-per740-68-vm-04 robottelo]$  pytest ./tests/upgrades/test_virtwho.py  -m post_upgrade --disable-pytest-warnings
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.9, pytest-7.3.2, pluggy-1.0.0
Mandatory Requirements are up to date.
Optional Requirements are up to date.
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/virtwho/repository/robottelo
configfile: pyproject.toml
plugins: cov-4.1.0, ibutsu-2.2.4, xdist-3.3.1, mock-3.11.1, services-2.2.1, reportportal-5.1.9
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_virtwho.py .                                                                                                                                                                            [100%]

================================================================================== 1 passed, 1 deselected, 9 warnings in 11.21s ===================================================================================

```

### CODE REVIEW

File: tests/upgrades/test_virtwho.py [hunk 1]
```diff
@@ -78,6 +78,9 @@ def test_pre_create_virt_who_configuration(
             3. Report is sent to satellite.
             4. Virtual sku can be generated and attached.
         """
+        org = target_sat.api.Organization().search(query={'search': f'name={ORG_DATA["name"]}'})
+        if org:
+            target_sat.api.Organization(id=org[0].id).delete()
```
Comment (omkarkhatavkar) [inline tests/upgrades/test_virtwho.py:83]: Another approach would be passing the dynamic org as part of pre_upgrade and adding that part of save_data. Non-blocking though

Comment (yanpliu) [reply tests/upgrades/test_virtwho.py:83]: > Another approach would be passing the dynamic org as part of pre_upgrade and adding that part of save_data. Non-blocking though

@omkarkhatavkar when use dynamic name 
```
ORG_DATA = {'name':gen_string('alpha')}
```
the post-upgrade will also use another dynamic name, this is not the same as the pre-upgrade case.
What is your suggestion?

Comment (omkarkhatavkar) [reply tests/upgrades/test_virtwho.py:83]: you can pass the dynamic org name in save_data function, fetch that same name in post upgrade

Comment (Gauravtalreja1) [inline tests/upgrades/test_virtwho.py:83]:
```suggestion
            org.delete()
```

Comment (yanpliu) [reply tests/upgrades/test_virtwho.py:83]: @omkarkhatavkar Have already use the dynamic name and save_test_data to realize this function, thanks for your help.

Comment (yanpliu) [reply tests/upgrades/test_virtwho.py:83]: > 
@Gauravtalreja1 Thank you very much for your good suggestion, This code has been replaced by dynamic name, so no need to delete the exist org.

**Review by shubhamsg199 — APPROVED** [review]

File: tests/upgrades/test_virtwho.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/upgrades/test_virtwho.py:83]: Yes, still instead of searching with API, we could use `org.delete()`

**Review by synkd — APPROVED** [review]

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**